csp/build/csp_autogen.py

```diff
@@ -387,6 +387,9 @@ def generate_cpp_code(self):
 {assertions}
     if( Py_IsInitialized() )
     {{
+        //Note that windows requires we grab the GIL since the windows DLL loading code releases GIL
+        PyGILState_STATE state = PyGILState_Ensure();
```

Comment:
We can use `csp::python::AcquireGIL` class here (we're just acquiring/releasing in scope).